mountpoint-s3/src/inode/readdir.rs

```diff
@@ -353,7 +353,7 @@ impl RemoteIter {
                     self.full_path.as_str(),
                 )
                 .await
-                .map_err(|e| InodeError::ClientError(anyhow::Error::new(e)))?;
+                .map_err(|e| InodeError::client_error(e, None, &self.bucket, &self.full_path))?;
```

Comment:
tiny nit: should we just add a context string here as well? And simply make it non-optional?